### DISCUSSION TITLE
[v5] requiredIfNotExists rule with strange behavior

### DISCUSSION THREAD

**Discussion by satheler** [adonisjs/core#1724] [Help]
I have this `Validating HTTP`

```typescript
const addressSchema = schema.create({
    user_id: schema.string.optional({ trim: true }, [
      rules.exists({ table: 'users', column: 'id' }),
      rules.requiredIfNotExists('company_id'),
    ]),

    company_id: schema.string.optional({ trim: true }, [
      rules.exists({ table: 'companies', column: 'id' }),
      rules.requiredIfNotExists('user_id'),
    ])

    description: schema.string({ trim: true }, []),
})
```

So I expected that either I would provide the `user_id` or the `company_id`, but not the 2 together. In practice, I can send both information and they are accepted.

Is this right? Or do I have to use another strategy?

**Comment by thetutlage** [comment]
Nope. `requiredIfNotExists` enforces a property to exist when the other one is missing. However, it doesn't check for exclusivity that only one should exist.
